Check gripper position in WsgAction.

avalenzu opened the pull request:
Now if the WSG fails to respond to a command, the `WsgAction` will recognize that it is not finished. Thanks to @sammy-tri for the patch.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/robotlocomotion/drake/7392)
<!-- Reviewable:end -->

avalenzu commented:
+@naveenoid for feature review.

---

Review status: 0 of 2 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7392#-:-KxwF-r40hsp9ejboDvw:b-ijjxgm)*
<!-- Sent from Reviewable.io -->

jamiesnape commented:
@drake-jenkins-bot linux-xenial-clang-cmake-experimental please
@drake-jenkins-bot linux-xenial-gcc-cmake-experimental-matlab please

naveenoid commented:
Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/action.h, line 147 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7392#-Ky1ZKc3kKvtiS7KvMgu:-Ky1ZKc3kKvtiS7KvMgv:bwbkqo0) ([raw file](https://github.com/robotlocomotion/drake/blob/133da690f57956a7161bb699b69ff6bfd0cbe767/drake/examples/kuka_iiwa_arm/pick_and_place/action.h#L147)):*
> ```objc
>   /**
>    * Returns true if the gripper stopped moving, and it is at least 0.5 seconds
>    * after an Open / Close command was last issued.
> ```

The open position threshold needs documentation somewhere. Ideally, I would prefer if its a class constant and passed as a default argument in constructor. There is a case to be made for modifying it if we add stuff to the gripper (for instance : gelsight). 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7392)*
<!-- Sent from Reviewable.io -->

sammy-tri commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/action.h, line 147 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7392#-Ky1ZKc3kKvtiS7KvMgu:-Ky21WyqFPfDcuzOOou8:bh6qd4m) ([raw file](https://github.com/robotlocomotion/drake/blob/133da690f57956a7161bb699b69ff6bfd0cbe767/drake/examples/kuka_iiwa_arm/pick_and_place/action.h#L147)):*
<details><summary><i>Previously, naveenoid (Naveen Kuppuswamy) wrote…</i></summary><blockquote>

The open position threshold needs documentation somewhere. Ideally, I would prefer if its a class constant and passed as a default argument in constructor. There is a case to be made for modifying it if we add stuff to the gripper (for instance : gelsight). 
</blockquote></details>

BTW IMO A default argument would be, currently, specified by no callers to the constructor.  Since it would have no effect until source code was modified and recompiled anyway, we could wait to add default arguments to the constructor until we know it would actually be useful.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7392)*
<!-- Sent from Reviewable.io -->

naveenoid commented:
Review status: all files reviewed at latest revision, all discussions resolved.

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/action.h, line 147 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7392#-Ky1ZKc3kKvtiS7KvMgu:-Ky2EL0Gq3vKaP_7-g_l:b-p1rw3c) ([raw file](https://github.com/robotlocomotion/drake/blob/133da690f57956a7161bb699b69ff6bfd0cbe767/drake/examples/kuka_iiwa_arm/pick_and_place/action.h#L147)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

BTW IMO A default argument would be, currently, specified by no callers to the constructor.  Since it would have no effect until source code was modified and recompiled anyway, we could wait to add default arguments to the constructor until we know it would actually be useful.
</blockquote></details>

Agreed with @sammy-tri. Will be happy with some documentation for now.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7392)*
<!-- Sent from Reviewable.io -->

avalenzu commented:
Addressed review comments. PTAL.

---

Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/action.h, line 147 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7392#-Ky1ZKc3kKvtiS7KvMgu:-KyWJhVk_5_PIYHJQXZe:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/133da690f57956a7161bb699b69ff6bfd0cbe767/drake/examples/kuka_iiwa_arm/pick_and_place/action.h#L147)):*
<details><summary><i>Previously, naveenoid (Naveen Kuppuswamy) wrote…</i></summary><blockquote>

Agreed with @sammy-tri. Will be happy with some documentation for now.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7392#-:-KyWJe9ljm7skvkX5Vnj:b4sixdv)*
<!-- Sent from Reviewable.io -->

naveenoid commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 2 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7392#-:-KyWLQ0yh48fo0n0mDOu:bnfp4nl)*
<!-- Sent from Reviewable.io -->

soonho-tri commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> (aside BTWs)

---

Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/action.h, line 147 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7392#-KyWeaQMWMbPGNQYL6HU:-KyWeaQNsp1ZWLNi1OZa:bt31b0z) ([raw file](https://github.com/robotlocomotion/drake/blob/4e67e0359599f73781ce671ea816cd73e62d4dc2/drake/examples/kuka_iiwa_arm/pick_and_place/action.h#L147)):*
> ```objc
>   /**
>    * Returns true if the following criteria are satisfied:
>    *  - The gripper velocity is less than the final velocity threshold
> ```

BTW, we check `|wsg_v| < threshold` in the code. Do we need to refine the comment to match it with the code?

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/action.h, line 156 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7392#-KyWerDmtXqvayRnuiak:-KyWerDmtXqvayRnuial:b-oejzep) ([raw file](https://github.com/robotlocomotion/drake/blob/4e67e0359599f73781ce671ea816cd73e62d4dc2/drake/examples/kuka_iiwa_arm/pick_and_place/action.h#L156)):*
> ```objc
>  private:
>   enum { kOpen, kClose } last_command_{kOpen};
>   double final_velocity_threshold_ = 1e-2;  // m/s
> ```

BTW, it seems that they are constants. If so, consider the following?
```c++
  static constexpr double kFinalVelocityThreshold = 1e-2;  // m/s
  static constexpr double kOpenPositionThreshold = .095;   // m
```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7392#-:-KyWfRVrk8a8AvZBCx_X:b5u6cca)*
<!-- Sent from Reviewable.io -->

avalenzu commented:
Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions.

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/action.h, line 147 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7392#-KyWeaQMWMbPGNQYL6HU:-KyXJhb0ZtLU-YkzTyah:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/4e67e0359599f73781ce671ea816cd73e62d4dc2/drake/examples/kuka_iiwa_arm/pick_and_place/action.h#L147)):*
<details><summary><i>Previously, soonho-tri (Soonho Kong) wrote…</i></summary><blockquote>

BTW, we check `|wsg_v| < threshold` in the code. Do we need to refine the comment to match it with the code?
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/action.h, line 156 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7392#-KyWerDmtXqvayRnuiak:-KyXJhAwz3hIGTod-sJW:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/4e67e0359599f73781ce671ea816cd73e62d4dc2/drake/examples/kuka_iiwa_arm/pick_and_place/action.h#L156)):*
<details><summary><i>Previously, soonho-tri (Soonho Kong) wrote…</i></summary><blockquote>

BTW, it seems that they are constants. If so, consider the following?
```c++
  static constexpr double kFinalVelocityThreshold = 1e-2;  // m/s
  static constexpr double kOpenPositionThreshold = .095;   // m
```
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7392)*
<!-- Sent from Reviewable.io -->

soonho-tri commented:
Reviewed 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/action.h, line 156 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7392#-KyWerDmtXqvayRnuiak:-KyXS--5Y3cR8AcBzgzt:b1prgir) ([raw file](https://github.com/robotlocomotion/drake/blob/4e67e0359599f73781ce671ea816cd73e62d4dc2/drake/examples/kuka_iiwa_arm/pick_and_place/action.h#L156)):*
FYI, we need to rename those threshold variables.

https://google.github.io/styleguide/cppguide.html#Constant_Names

> All such variables with static storage duration (i.e. statics and globals, see Storage Duration for details) should be named this way.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7392)*
<!-- Sent from Reviewable.io -->

avalenzu commented:
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/action.h, line 156 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7392#-KyWerDmtXqvayRnuiak:-KyXbwkwR6YnKGo0l79M:bu9g4kj) ([raw file](https://github.com/robotlocomotion/drake/blob/4e67e0359599f73781ce671ea816cd73e62d4dc2/drake/examples/kuka_iiwa_arm/pick_and_place/action.h#L156)):*
<details><summary><i>Previously, soonho-tri (Soonho Kong) wrote…</i></summary><blockquote>

FYI, we need to rename those threshold variables.

https://google.github.io/styleguide/cppguide.html#Constant_Names

> All such variables with static storage duration (i.e. statics and globals, see Storage Duration for details) should be named this way.
</blockquote></details>

Done. Thanks for catching that.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7392)*
<!-- Sent from Reviewable.io -->

soonho-tri commented:
Reviewed 2 of 2 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7392)*
<!-- Sent from Reviewable.io -->

avalenzu commented:
@drake-jenkins-bot linux-xenial-clang-bazel-experimental-debug please

---

Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7392#-:-Ky_uAaDGjUiAwiNh6z-:bwc74yn)*
<!-- Sent from Reviewable.io -->
